File: GAP-FrontEnd/src/app/services/Interfaces/affectation-request-dto.ts
import {Iemploye} from "./iemploye";
import {Iprojet} from "./iprojet";
import {Iateliers} from "./iateliers";
import {Iarticle} from "./iarticle";

export interface AffectationRequestDTO {

  date: Date;
  periode: string;
  nombreHeures: number;
  employees: Iemploye[];
  projets: Iprojet;
  ateliers: Iateliers;
  article: Iarticle;
}

File: GAP-BAckEnd/src/main/java/ma/gap/dtos/AffectationRequestDTO.java
package ma.gap.dtos;

import lombok.Data;
import ma.gap.entity.*;
import java.util.Date;
import java.util.List;

@Data
public class AffectationRequestDTO {
    private Date date;
    private String periode;
    private Integer nombreHeures;
    private List<Employee> employees;
    private Projet projets;
    private Ateliers ateliers;
    private Article article;
}